perllib/FixMyStreet/Cobrand/Oxfordshire.pm

```diff
@@ -284,4 +289,98 @@ sub dashboard_export_problems_add_columns {
     });
 }
 
+sub defect_wfs_query {
+    my ($self, $params) = @_;
+
+    my $uri = URI->new("https://tilma.mysociety.org/proxy/occ/nsg/");
+    $uri->query_form(
+        %$params,
+        REQUEST => "GetFeature",
+        SERVICE => "WFS",
+        SRSNAME => "urn:ogc:def:crs:EPSG::4326",
+        TYPENAME => "WFS_DEFECTS_FOR_QUERYING",
+        VERSION => "1.1.0",
+        propertyName => 'ITEM_CATEGORY_NAME,ITEM_TYPE_NAME,REQUIRED_COMPLETION_DATE,ACTUAL_COMPLETION_DATE,APPROVAL_STATUS_NAME,SHAPE_GEOMETRY',
+        outputformat => "application/json"
+    );
+
+    try {
+        my $response = get($uri);
+        my $json = JSON->new->utf8->allow_nonref;
+        return $json->decode($response);
+    } catch {
+        # Ignore WFS errors.
+        return [];
+    };
+}
+
+# Get defects from WDM feed and display them on /around page.
+sub pins_from_wfs {
+    my ($self, $bbox) = @_;
+
+    my $wfs = $self->defect_wfs_query({
+        BBOX => $bbox,
+    });
+
+
+    my @features = grep {
+        $_->{properties}->{APPROVAL_STATUS_NAME} eq 'With Contractor';
```

Comment:
WFS allows server-side filtering of features which would remove the need for this grep (and hopefully speed up the WFS response time too). If you added a `filter` param to the WFS request that looked something like this it'd have the desired effect, I think:

```xml
<ogc:Filter xmlns:ogc="http://www.opengis.net/ogc">
    <ogc:And>
        <ogc:PropertyIsEqualTo matchCase="true">
            <ogc:PropertyName>APPROVAL_STATUS_NAME</ogc:PropertyName>
            <ogc:Literal>With Contractor</ogc:Literal>
        </ogc:PropertyIsEqualTo>
        <ogc:BBOX>
            <ogc:PropertyName>SHAPE_GEOMETRY</ogc:PropertyName>
            <gml:Envelope xmlns:gml="http://www.opengis.net/gml" srsName="EPSG:4326">
                <gml:lowerCorner>-1.3134050585937 51.80585466836</gml:lowerCorner>
                <gml:upperCorner>-1.2090349414063 51.84765563379</gml:upperCorner>
            </gml:Envelope>
        </ogc:BBOX>
    </ogc:And>
</ogc:Filter>
```

Might have to remove the newlines and other whitespace :) `filter` is mutually exclusive with the `BBOX` param, hence it being included within the `<And>` in the filter. You could lose `APPROVAL_STATUS_NAME` from the `propertyName` param too.

Reply:
Oh this is nice, thanks for the tip-off!

Done in 276f841b24e7dc947cc14b0cbb38491872aa7a2b